Prep manuscript for resubmission

Nikoleta-v3 opened the pull request.

Commit:
adds files for scientific reports templete

Commit:
add tex files to submission folder

Commit:
add images to submission folder

Commit:
add bibliography to submission folder

Commit:
initial draft of the paper in templete

Commit:
removes ^T from stability of defection lemma

Commit:
initial draft of supplementary information

Commit:
s/Eq./Equation and adds paranthesis

Commit:
s/Fig./Figure

Commit:
compile pdfs

Commit:
add cover letter to response

Nikoleta-v3 commented:
For your response we had to:

>>> In order to process your paper, we require the following:

 >>> • A cover letter describing your response to our editorial requests.
 >>> • A point-by-point response to any issues raised by the reviewers. This must be uploaded as ‘Related file’. Our guide to writing a response to the reviewers’ comments provides advice for maximising its effectiveness.

That's why I added a cover letter in the response file. https://github.com/Nikoleta-v3/Memory-size-in-the-prisoners-dilemma/pull/63/commits/da7866d74d4635406be87766c07ffc07cd8cf138

drvinceknight commented:
Everything looks good @Nikoleta-v3, just the diff'd pdf would be ideal (it's essentially part of the response to reviewers0.

Commit:
move theorem to main body and rename appendices

This has been done to fit the new submition to scientific reports

Commit:
points that the proof of theorem 2 is in the SI

Commit:
generate diff tex

Commit:
add diff file to response

drvinceknight committed:
Write optimal moran player experiments (#69) (#74)

* Adjust the environment due to upstream updates

Tests would not run on my machine due to some upstream change in:

- `matplotlib`, error messages lead me to:
  https://stackoverflow.com/questions/31373163/anaconda-runtime-error-python-is-not-installed-as-a-framework/41433353
- `sklearn` (a dependency of `scikit-opt`), error messages lead me to:
  https://github.com/skggm/skggm/issues/124

* Write tests for ability for K self interactions

* Implement ability to have K self interactions

@Nikoleta-V3 could you take a close look at the tests please. I'm not
entirely sure the behaviour makes sense to me (I would expect that as K
increases the optimal behaviour would be to in general cooperate
more...).

* Add data structure for data collection.

Looks like `data/` is in `.gitignore` so I've called this `moran_data`
for now but we should find a better data structure.

Also: I'm writing to `csv` it would be nicer to write to an sql
database.

* Rewrite README to use analytical approach.

* Remove unnecessary line.

* Add computation of non dynamic fixation.

* Write script to carry out Moran process.

* Reduce absolute tolerance for two numeric tests.

* Run black.

* Run black again.

* Apply suggestions from code review with Nik

* rename function and also return cycle

renames get_repeat_length_in_history to get_repeat_cycle_and_length

* uncomment tests

* return the strategy with the highest score in case of a cycle

the tests are currently failing (correctly)

* pass tests

* Fix typos in README

* Update src/opt_mo/evolutionary_best_response.py

Co-authored-by: Vince Knight <vincent.knight@gmail.com>

Co-authored-by: Nikoleta Glynatsi <glynatsine@cardiff.ac.uk>

Co-authored-by: Nikoleta Glynatsi <glynatsine@cardiff.ac.uk>

drvinceknight committed:
Add analysis of moran process (#73)

* Add a notebook.

* Plot figure.

* Add discussion of Moran process.

* Run black.

* Make minor suggestions.

Co-authored-by: Nikoleta Glynatsi <GlynatsiNE@cardiff.ac.uk>

* Add `where`

Co-authored-by: Nikoleta Glynatsi <GlynatsiNE@cardiff.ac.uk>

* Update results with new data.

* uses capital K consistently over the manuscript

* suggests structure

I am happy for the results section to remain as it is. I suggest
including some subsection will help with the narative.

The suggested titles are just suggestios.
I am happy to un-do the commit.
The moran section still needs tweaking (which includes adding discussion in the conclusion)

* Reorder the results section.

This now goes:

S. Results

s. Best response
s. Best response against a group
s. Limitations of extortion
s. Limitations of memory
s. Dynamic best response player

* Fix Moran process part of results section.

* s/N/n in Moran process.

* Adjust discussion.

* removes discussion on self interactions

* corrects minor stuff to compile document

Co-authored-by: Nikoleta Glynatsi <GlynatsiNE@cardiff.ac.uk>

drvinceknight committed:
Fix ordering of data generation script. (#76)

Commit:
Addressing minor comment from the reviewers (#77)

* moves "stability of defection" to appendix

* fixes citation style

* delete previous response to reviewers

* adds response to reviewers

These are just notes

* adds discussion on theory of mind literature

* adds equations with noise

* adds to the response

* compile

Co-authored-by: Nikoleta-v3 <glynatsi@dynosob003.evolbio.mpg.de>

Commit:
Read through (#78)

* small tweaks

* last rewrites by n.g.

* Make minor changes

Take a look at these @nikoleta-v3 as I specifically remove one addition you made - we can chat about that during our meeting (the other changes I'm just fixing the language).

* Update figure with latest data.

* Add some summary statistics.

* Update main.tex

Co-authored-by: Vince Knight <vincent.knight@gmail.com>

* Update main.tex

Co-authored-by: Vince Knight <vincent.knight@gmail.com>

* Update data and fix minor typos.

* Fix citations (#79)

* Add data citation

* include citations to data to the main.tex

Co-authored-by: Vince Knight <vincent.knight@gmail.com>

drvinceknight committed:
Write response for scientific reports (#80)

* Make some tweaks to response.

Specifically add discussion about Moran process addition.

* Remove TODO from cover letter.

* Compile the pdfs.

Commit:
updates figures and renames them to Fig 1 - 4

Commit:
add affiliation to me and fixes caption

In the caption tournaments with no self interactions were being
mentioned.

Commit:
apply all the revision changes to the templeted manuscript

Commit:
move appendix material to supplementary information

Commit:
refer to SI instead of the Appendix + last minor tweaks to fit style

Commit:
generates new diff.pdf

Commit:
fix style for refering to equations

Commit:
fix references style

Commit:
change manuscript title to Using a theory of mind to find best responses to memory-one strategies.

Commit:
removes figures that are not included in the current version

Commit:
compile SI
